tests/tensor/rewriting/test_shape.py

```diff
@@ -476,6 +477,30 @@ def test_vector_dim_err(self):
             shape_feature.same_shape(x, o, 0, 1)
 
 
+def test_useless_specify_shape():
+    x = tensor("x", shape=(None, 5, 3))
+
+    # We avoid the helper specify_shape that optimizes some (but not all) cases eagerly
+    ss = SpecifyShape()
+
+    out = ss(x, None, 5, None)
+    assert isinstance(out.owner.op, SpecifyShape)
+    ret = local_useless_specify_shape.transform(None, out.owner)
+    assert ret == [x]
+
+    # SpecifyShape is needed to enfore unknown dim is 3
+    out = ss(x, 3, 5, None)
+    assert isinstance(out.owner.op, SpecifyShape)
+    ret = local_useless_specify_shape.transform(None, out.owner)
+    assert ret is None
+
+    # SpecifyShape is needed to raise mismatch between static and specified dim
+    out = ss(x, None, 5, 4)
```

Comment:
Beyond the scope of this PR, but `SpecifyShape` could just eagerly raise in this case?

Reply:
Yes, there's also a config flag to raise during rewrites when shapes are found to be incompatible in the shape feature. But I don't see why not do it all the time.

Can you open an issue?